src/main/java/com/zendesk/maxwell/MaxwellReplicator.java

```diff
@@ -134,7 +134,11 @@ private MaxwellAbstractRowsEvent processRowsEvent(AbstractRowEvent e) throws Sch
 
 		table = tableCache.getTable(e.getTableId());
 
-		if (table == null) {
+		if ( table == null && this.filter.hasTableBlacklist() ) {
+			LOGGER.warn(String.format("couldn't find table in cache for table id: %d, assuming blacklisted table", e.getTableId()));
+			return null;
```

Comment:
Yeah, I thought you might say that :)
If you want to remember the mapping of `tableId` to `tableName`, then you'd have to persist it in the `maxwell` schema, correct? An in-memory `Set` wouldn't be enough I think.